core/src/main/java/com/homeaway/streamplatform/streamregistry/service/impl/StreamServiceImpl.java

```diff
@@ -0,0 +1,247 @@
+        List<String> vpcList = stream.getVpcList();
+        List<String> replicatedVpcList = stream.getReplicatedVpcList();
+        log.info("creating topics for vpcList: {}", vpcList);
+        for (String vpc : vpcList) {
+            upsertTopics(stream, vpc, isNewStream);
+        }
+        if (replicatedVpcList != null && !replicatedVpcList.isEmpty()) {
+            log.info("creating topics for replicatedVpcList: {}", replicatedVpcList);
+            for (String vpc : replicatedVpcList) {
+                upsertTopics(stream, vpc, isNewStream);
+            }
+        }
+    }
+
+    private void upsertTopics(Stream stream, String vpc, boolean isNewStream) throws StreamCreationException, ClusterNotFoundException {
+        ClusterValue clusterValue = clusterService.getCluster(vpc, env, stream.getTags().getHint(), "producer");
+        Properties topicConfig = new Properties();
+        if (stream.getTopicConfig() != null) {
+            topicConfig.putAll(stream.getTopicConfig());
+        }
+
+        String bootstrapServer = clusterValue.getClusterProperties().get(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG);
+        String zkConnect = clusterValue.getClusterProperties().get(KafkaProducerConfig.ZOOKEEPER_QUORUM);
+        topicConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServer);
+        topicConfig.put(KafkaProducerConfig.ZOOKEEPER_QUORUM, zkConnect);
+
+        kafkaManager.upsertTopics(Collections.singleton(stream.getName()), stream.getPartitions(), stream.getReplicationFactor(), topicConfig, isNewStream);
+        log.info("Topic {} created/updated at {}", stream.getName(), bootstrapServer);
+    }
+
+    private void applyDefaultHint(Stream stream) {
+        Tags tags = stream.getTags();
+
+        String hint = tags.getHint();
+
+        if (hint == null || hint.trim().matches("(?i:string)?")) {
+            hint = AbstractService.PRIMARY_HINT;
+        }
+
+        tags.setHint(hint);
+    }
+
+    @Override
+    public Stream getStream(String streamName) throws StreamNotFoundException {
+        log.info("Pulling stream information from global state-store for streamName={}", streamName);
+        Optional<AvroStream> streamValue = streamDao.getStream(streamName).getValue();
+
+        if (!streamValue.isPresent()) {
+            throw new StreamNotFoundException(String.format("Stream=%s not found.", streamName));
+        } else {
+            return AvroToJsonDTO.convertAvroToJson(streamValue.get());
+        }
+    }
+
+    @Override
+    public void deleteStream(String streamName) throws StreamNotFoundException{
+        Pair<AvroStreamKey, Optional<AvroStream>> keyValue = streamDao.getStream(streamName);
+
+        if (!keyValue.getValue().isPresent()) {
+            throw new StreamNotFoundException(String.format("Stream=%s not found.", streamName));
+        }
+        try {
+            streamDao.upsertStream(keyValue.getKey(), null);
```

Comment:
Do you really want the service to swallow all exceptions ? Shouldn't runtime exceptions be bubbled up to the resource layer to be handled as 500s?

Reply:
When i refactored i missed it. 
Fixed it now. Throwing back all the Runtime Exceptions and Resource class is equipped to handle it properly